### PR TITLE
Wrote RailRoadSquare Class

### PR DESCRIPTION
**Pull request by kernohad**

### CODE REVIEW

File: Model/src/BoardPackage/RailRoadSquare.java [hunk 1]
```diff
@@ -1,7 +1,131 @@
...
+    /**The rent if a player owns four RailRoads*/
+    private final int FOUR_RENT;
+
+    /**The value a player receives if they mortgage this RailRoad*/
+    private final int MORTGAGE_VALUE;
+
+    /**A flag to tell if this RailRoad is currently mortgaged*/
+    private boolean isMortgaged;
+
+
```
Comment (Dcbaas) [inline Model/src/BoardPackage/RailRoadSquare.java]: Need an owner variable. We forgot that in the UML. Add related getter and setter.

Comment (kernohad) [reply Model/src/BoardPackage/RailRoadSquare.java]: Ah, yes. good catch. Will get to that in a bit.

File: Model/src/BoardPackage/RailRoadSquare.java [hunk 2]
```diff
@@ -1,7 +1,162 @@
...
+ The RailRoadSquare class is a BoardSquare that tracks
+ information specific to RailRoadSquares.
+
+ @author Dylan Kernohan
+ @version 2/12/2018
+ *****************************************************************/
+public class RailRoadSquare extends BoardSquare{
+
+    /**The price of the RailRoad square*/
+    private final int PRICE;
```
Comment (quirogas) [inline Model/src/BoardPackage/RailRoadSquare.java]: is it possible to call a constant and not instantiate it right away ? also can it even be instantiate in the constructor ?

File: Model/src/BoardPackage/RailRoadSquare.java [hunk 3]
```diff
@@ -1,7 +1,162 @@
...
+
+        super(name);
+        this.PRICE = price;
+        this.BASE_RENT = baseRent;
+        this.TWO_RENT = twoRent;
+        this.THREE_RENT = threeRent;
+        this.FOUR_RENT = fourRent;
+        this.MORTGAGE_VALUE = mortgageValue;
+        this.isMortgaged = false;
+        // TODO: Do we want to keep this null or make Bank own all squares to start out with? If so we would add Bank as param
```
Comment (quirogas) [inline Model/src/BoardPackage/RailRoadSquare.java]: Bank

Comment (quirogas) [inline Model/src/BoardPackage/RailRoadSquare.java]: remove this comment. Next time you can just do the comment through the PR instead of the actual code.

Comment (quirogas) [inline Model/src/BoardPackage/RailRoadSquare.java]: remove the commnet

**Comment by dustinecker** [comment]
Don't you implement interfaces instead of extending them or does it not matter for our BoardSquare class?

**Comment by kernohad** [comment]
BoardSquare is not an interface. We initially were thinking about making it one but just made it an abstract class. So it would be extends. I believe xD

**Pull request merged**